frontend/src/Editor/Viewer.jsx

```diff
@@ -550,6 +552,8 @@ class ViewerComponent extends React.Component {
     );
   };
 
+  setIsSidebarPinned = () => this.setState({ isSidebarPinned: !this.state.isSidebarPinned });
```

Comment:
Suggestion: a better name for this function is `toggleSidebarPinned`, since it toggles the state without an argument